File: 1534.py
arr = list(map(int, input().split()))
a = int(input())
b = int(input())
c = int(input())
count = 0

for i in range (0, len(arr)-2):
    for j in range (i+1, len(arr)-1):
        for k in range (j+1, len(arr)):
            if abs(arr[i] - arr[j]) <= a and abs(arr[j] - arr[k]) <= b and abs(arr[i] - arr[k]) <= c:
                count+=1

print(count)

File: 1534.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
    std::vector<int> arr = {3, 0, 1, 1, 9, 7};
    int a = 7, b = 2, c = 3;
    int count = 0;

    for (int i = 0; i < arr.size(); i++)
    {
        for (int j = i + 1; j < arr.size(); j++)
        {

            if (abs(arr[i] - arr[j]) <= a)
            {
                for (int k = j + 1; k < arr.size(); k++)
                {

                    if (abs(arr[j] - arr[k]) <= b && abs(arr[i] - arr[k]) <= c)
                    {
                        count++;
                    }
                }
            }
        }
    }

    cout << count;
}